packages/core/admin/admin/src/pages/SettingsPage/components/Tokens/Table/DeleteButton/tests/DeleteButton.test.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { ThemeProvider, lightTheme } from '@strapi/design-system';
+import { act, fireEvent, render } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import { NotificationsProvider } from '@strapi/helper-plugin';
+
+import DeleteButton from '../index';
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
+}));
+
+function getButton(container, name) {
+  return container.querySelector(`button[name="${name}"]`);
+}
```

Comment:
This is a really bad pattern imo, we have `getByRole` and you can pass options to utilise their name. This would also avoid you using `baseElement` in your tests. Because then the selectors are focussed on how a user uses the application, not the code.

Reply:
function removed